llama-index-integrations/vector_stores/llama-index-vector-stores-alibabacloud-mysql/llama_index/vector_stores/alibabacloud_mysql/base.py

```diff
@@ -0,0 +1,579 @@
+"""Alibaba Cloud MySQL Vector Store."""
+
+import json
+import logging
+from typing import Any, Dict, List, NamedTuple, Optional
+from contextlib import contextmanager
+
+import mysql.connector
+import mysql.connector.pooling
+from mysql.connector.connection import MySQLConnection
+from mysql.connector.cursor import MySQLCursor
+from mysql.connector.errors import Error as MySQLError
+
+from llama_index.core.bridge.pydantic import PrivateAttr
+from llama_index.core.schema import BaseNode, MetadataMode
+from llama_index.core.vector_stores.types import (
+    BasePydanticVectorStore,
+    FilterCondition,
+    FilterOperator,
+    MetadataFilter,
+    MetadataFilters,
+    VectorStoreQuery,
+    VectorStoreQueryMode,
+    VectorStoreQueryResult,
+)
+from llama_index.core.vector_stores.utils import (
+    metadata_dict_to_node,
+    node_to_metadata_dict,
+)
+
+
+class DBEmbeddingRow(NamedTuple):
+    node_id: str
+    text: str
+    metadata: dict
+    similarity: float
+
+
+_logger = logging.getLogger(__name__)
+
+
+class AlibabaCloudMySQLVectorStore(BasePydanticVectorStore):
+    """
+    Alibaba Cloud MySQL Vector Store.
+
+    Examples:
+        ```python
+        from llama_index.vector_stores.alibabacloud_mysql import AlibabaCloudMySQLVectorStore
+
+        # Create AlibabaCloudMySQLVectorStore instance
+        vector_store = AlibabaCloudMySQLVectorStore(
+            table_name="llama_index_vectorstore",
+            host="localhost",
+            port=3306,
+            user="llamaindex",
+            password="password",
+            database="vectordb",
+            embed_dim=1536,  # OpenAI embedding dimension
+            default_m=6,
+            distance_method="COSINE"
+        )
+        ```
+
+    """
+
+    stores_text: bool = True
+    flat_metadata: bool = False
+
+    table_name: str
+    host: str
+    port: int
+    user: str
+    password: str
+    database: str
+    charset: str
+    max_connection: int
+    embed_dim: int
+    default_m: int
+    distance_method: str
```

Comment:
Use `Literal["EUCLIDEAN", "COSINE"]` so that Pydantic automatically validates this on init and you don't have to validate it manually in the `_initialize`method:


```suggestion
    distance_method: Literal["EUCLIDEAN", "COSINE"]
```